### PR TITLE
[DDW-717] Change classnames from array to object

### PR DESCRIPTION
**Pull request by lrosa007**
Changes for #1434

This PR replaces the use of arrays by objects when using `classnames`

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by nikolaglumac** [comment]
@lrosa007 there are some flow errors - can you please make sure to run `yarn flow:test` and fix all the warnings? 🙏

**Comment by lrosa007** [comment]
@nikolaglumac I'll fix it today

**Comment by lrosa007** [comment]
@nikolaglumac @daniloprates 

Ok, I think I fixed everything. Sorry for all the excess commits

File: source/renderer/app/components/wallet/transactions/Transaction.js [hunk 1]
```diff
@@ -194,33 +194,26 @@ export default class Transaction extends Component<Props> {
...
+    });
 
-    const detailsStyles = classNames([
-      styles.details,
-      canOpenExplorer ? styles.clickable : null,
-      isExpanded ? styles.detailsExpanded : styles.detailsClosed,
-    ]);
+    const detailsStyles = classNames(styles.details, {
+      [styles.clickable]: canOpenExplorer,
+      [styles.detailsExpanded]: isExpanded,
```
Comment (nikolaglumac) [inline source/renderer/app/components/wallet/transactions/Transaction.js]: @lrosa007 this line is wrong... It should read:
```
[styles.detailsExpanded]: isExpanded,
[styles.detailsClosed]: !isExpanded,
```

**Comment by IuliaDolishniak** [comment]
Issues found at first glance:
1. Transactions cannot be collapsed
screen
2. 
screen2
3. 
screen2 (1)

File: source/renderer/app/components/wallet/transactions/Transaction.js (hunk 1 above)
Comment (lrosa007) [reply source/renderer/app/components/wallet/transactions/Transaction.js]: Thanks, fixing ASAP

**Comment by lrosa007** [comment]
@IuliaDolishniak I think that's related to the issue that @nikolaglumac commented about. Thank you 👍

File: source/renderer/app/components/wallet/transactions/Transaction.js (hunk 1 above)
Comment (lrosa007) [reply source/renderer/app/components/wallet/transactions/Transaction.js]: @nikolaglumac ok it's addressed

**Comment by lrosa007** [comment]
@IuliaDolishniak @nikolaglumac 

I've pushed a fix for that.

**Comment by daniloprates** [comment]
@lrosa007 we apologize, but we are currently going through a fast-paced development, due to the 
Cardano Incentivized Testnet, and these changes ended up being too costly for us to test, as they surfaced the entire app.

**Comment by lrosa007** [comment]
@daniloprates no worries. I figured that was the cause because I’ve been keeping up with the updates 🙂. Hopefully it’s useful for future reference. The idea is better done little by little anyways. There were already some instances of using classnames in the way I described anyways.

**Pull request closed without merging**